### ISSUE TITLE
[feat request] Ability to submit notifications from elsewhere

### ISSUE THREAD

**Issue by derekdreery** [linebender/druid#2093]
https://github.com/linebender/druid/blob/2f5beb8faba648c170a485b8d6a00d0d991a1737/druid/src/contexts.rs#L542

Could this method be added to other contexts, specifically `LifecycleCtx` and `UpdateCtx`? I want to submit a notification when the size of the widget changes or when the contents change.

**Comment by xarvic** [comment]
The problem is, handling a notification is an event. 
A workaround is to send a `Command` to this widget which then submits a `Notification` with the same Selector and payload.
We could automate this by adding a `forward_as_notification` flag to `Command`.
Another solution is to make `Notification` part of `LifeCycle`, since you can call LifeCycle from all of the other contexts, but this is a breaking change and only possible if there is no widget which updates data from `Event::Notification`

**Comment by xarvic** [comment]
There are widgets which change data in response to a `Notification`. These `Notification`s have to be send as `Event`s. 

Maybe we could send every `Notification` which was submitted during `Widget::event`, as an `Event` and every other `Notification` as an `Lifecycle`? The problem is, this might get confusing, if you are checking for `Notification`s in `Widget::event` but the `Notification` is send as an `Lifecycle`.

Another option is to add a new kind of `Notification`.

**Comment by derekdreery** [comment]
Yeah it's not easy. I'm thinking that eventually the whole data model of druid will be experimented upon in the future (after the piet/piet-gpu foundations are solid). I don't want anyone to spend lots of time on my niche use case in the mean time.
